src/casp/services/api/services/cell_operations_service.py

```diff
@@ -68,28 +72,24 @@ async def get_embeddings(file_to_embed: t.BinaryIO, model_name: str) -> t.Tuple[
         embeddings = numpy_utils.base64_to_numpy(embeddings_response_json["embeddings_b64"])
         return query_ids, embeddings
 
-    def __get_match_index_endpoint_client_for_model(
-        self, model_name: str
-    ) -> t.Tuple[models.CASMatchingEngineIndex, clients.CustomMatchingEngineIndexEndpointClient]:
-        index = self.cellarium_general_dm.get_index_for_model(model_name=model_name)
-
-        return index, clients.CustomMatchingEngineIndexEndpointClient(index_endpoint_name=index.endpoint_id)
+    def __get_match_index_endpoint_for_model(self, model_name: str) -> models.CASMatchingEngineIndex:
+        return self.cellarium_general_dm.get_index_for_model(model_name=model_name)
 
     @staticmethod
-    def __validate_knn_response(
-        embeddings: np.array, knn_response: t.Union[t.List[t.List[MatchNeighbor]], t.List[t.List[t.Dict[str, t.Any]]]]
-    ) -> None:
-        if len(knn_response) != len(embeddings):
+    def __validate_knn_response(embeddings: np.array, knn_response: MatchResult) -> None:
+        if len(knn_response.matches) != len(embeddings):
             raise exceptions.VectorSearchResponseError(
-                f"Number of query ids ({len(embeddings)}) and knn matches ({len(knn_response)}) does not match. "
+                f"Number of query ids ({len(embeddings)}) and knn matches ({len(knn_response.matches)}) does not match. "
                 f"This could probably be caused by Vector Search overload."
             )
-        if len(knn_response[0]) == 0:
-            raise exceptions.VectorSearchResponseError("Vector Search returned a match with 0 neighbors.")
 
-    def __get_knn_matches_with_retry(
+        for match in knn_response.matches:
```

Comment:
Is this a possible error mode we weren't accounting for before, or is this something specific to the switch to using MatchResult?

Reply:
Yeah, that's what it looks like.  We were only looking for the first match to not have neighbors.  It seemed like a better idea to make sure every single match had neighbors. 
 @fedorgrab does that seem right?

Reply:
I've never seen such cases, that if there was only one missing (basically I saw only results where either all of them were empty arrays, or none).

However, being extra cautious doesn't seem like a bad thing to me, especially if it doesn't increase the runtime, which I believe is the case.

Also, if we're going there, might be a good idea to check if the length of the nearest neighbors array equals to the embedding dimension, not just comparing it with a zero.